synapse/tests/test_lib_types.py

```diff
@@ -1,14 +1,67 @@
 # -*- coding: UTF-8 -*-
 import base64
 
-import synapse.cores.common as s_cores_common
 
 import synapse.common as s_common
 import synapse.lib.types as s_types
+import synapse.lib.xact as s_xact
 
 from synapse.tests.common import *
 
-class DataTypesTest(SynTest):
+
+class TestTypes(SynTest):
+
+    def test_hex_type(self):
+        with self.getTestCore() as core:
+            # fixme - getTestCore should have this dude loaded in him already!
+            modu = core.addCoreMods([('synapse.tests.test_cortex.TestModule', {})])
+
+            t = core.model.type('testhexa')
+            # Test norming & index
+            testvectors = [
+                ('C', 'c', b'\x0c'),
+                ('10', '10', b'\x10'),
+                ('10001', '10001', b'\x01\x00\x01'),
+                ('0x10001', '10001', b'\x01\x00\x01'),
+                ('FfF', 'fff', b'\x0f\xff'),
+                ('12A3e', '12a3e', b'\x01\x2a\x3e'),
+                (65537, '10001', b'\x01\x00\x01'),
+                (0, '0', b'\x00'),
+                (b'\x01\x00\x01', '10001', b'\x01\x00\x01'),
+                (b'\xd4\x1d\x8c\xd9\x8f\x00\xb2\x04\xe9\x80\t\x98\xec\xf8B~',
+                 'd41d8cd98f00b204e9800998ecf8427e',
+                 b'\xd4\x1d\x8c\xd9\x8f\x00\xb2\x04\xe9\x80\t\x98\xec\xf8B~'),
+                (-10, s_exc.BadTypeValu, None),
```

Comment:
Should be able to handle negative hex values right? Leave this as a TODO like int?

Reply:
OK. Checked and it is also not supported in current Synapse.